test/functional/test_framework/test_framework.py

```diff
@@ -531,9 +531,25 @@ def add_dynamically_node(self, extra_args=None, *, rpchost=None, binary=None):
         return t_node
 
     def dynamically_initialize_datadir(self, node_p2p_port, node_rpc_port):
-        data_dir = get_datadir_path(self.options.tmpdir, len(self.nodes))
-        if not os.path.isdir(data_dir):
-            os.makedirs(data_dir)
+        source_data_dir = get_datadir_path(self.options.tmpdir, 0)  # use node0 as a source
+        new_data_dir = get_datadir_path(self.options.tmpdir, len(self.nodes))
+
+        # In general, it's a pretty bad idea to copy datadir folder on the fly...
+        # But we flush all state changes to disk via gettxoutsetinfo call and
+        # we don't care about wallets, so it works
+        self.nodes[0].gettxoutsetinfo()
+        shutil.copytree(source_data_dir, new_data_dir)
+
+        def new_chain_path(*paths):
+            chain = get_chain_folder(new_data_dir, self.chain)
+            return os.path.join(new_data_dir, chain, *paths)
+
+        shutil.rmtree(new_chain_path('wallets'))
```

Comment:
`shutil.rmtree` that calls some function doesn't look for me as an error-prune solution.

Someone later can change the function `new_chain_path` somehow and make a bug that may be a result of removing his `${HOME}` directory with all data just by mistake during debugging and writing code.

To prevent it, I'd rewrite that code a bit:

```
new_chain_path = get_chain_folder(new_data_dir, self.chain)
shutil.rmtree(os.path.join(new_chain_path, 'wallets'))
shutil.rmtree(os.path.join(new_chain_path, 'llmq'))
```

Similar usages of `rmtree` are in `test/functional/wallet_backup.py`, `test/functional/wallet_hd.py` and other places and most of them show clear intention which directories would be deleted:
```
test/functional/wallet_backup.py:        shutil.rmtree(os.path.join(self.nodes[2].datadir, self.chain, 'blocks'))
test/functional/test_runner.py:        shutil.rmtree("%s/test/cache" % config["environment"]["BUILDDIR"], ignore_errors=True)
```

I think that's potentially risky code and `rmtree` should not accept any result of function such as `new_chain_path` that can be changed without acknowledgement what exactly can be a consequence.

Benefits of "avoiding code duplication" as a spare function doesn't out-weight potential danger of this code.